be/src/io/hdfs_util.cpp

```diff
@@ -103,16 +103,16 @@ void HdfsHandlerCache::_clean_oldest() {
 }
 
 Status HdfsHandlerCache::get_connection(const THdfsParams& hdfs_params, const std::string& fs_name,
-                                        HdfsHandler** fs_handle) {
+                                        std::shared_ptr<HdfsHandler>* fs_handle) {
     uint64 hash_code = hdfs_hash_code(hdfs_params);
     {
         std::lock_guard<std::mutex> l(_lock);
         auto it = _cache.find(hash_code);
         if (it != _cache.end()) {
-            HdfsHandler* handle = it->second.get();
+            std::shared_ptr<HdfsHandler> handle = it->second;
             if (!handle->invalid()) {
                 handle->inc_ref();
-                *fs_handle = handle;
+                *fs_handle = std::move(handle);
```

Comment:
why do we still have to deal with `inc_ref()` even with shared_ptr support? 
can we integrate the process (inc_ref, dec_ref) into  shared_ptr ?

Reply:
It is for fs handle cache.